packages/discussions/src/types.ts

```diff
@@ -80,7 +80,8 @@ export enum PostStatus {
 export enum DiscussionType {
   DATASET = "dataset",
   ITEM = "item",
-  GROUP = "group"
+  GROUP = "group",
+  CONTENT = "content",
```

Comment:
This enum is what is causing the [Discussions API](https://devtopia.esri.com/dc/hub-discussions/blob/main/src/posts/dto/validators/is-discussion-uri.ts#L19) to serve an error when we try to create a new post w/ a `content` URI, e.g. `hub://content/12345`. I question if we _really_ want to restrict this; we'll need to update this ENUM and release any time we want a new discussion URI pattern. The alternative would be to refactor the API to completely remove this restriction, then we can deprecate this ENUM.

Reply:
I agree with the concern about having to regularly update this enumeration for validation. 

Instead, what about using this enumeration just for string operations (e.g. devs only use the enumeration to create the URI) to prevent mistypings, but _not_ for actual validation. Similar pattern to the Telemetry Dictionaries as a method to prevent event string name mistypings. 

e.g. ``let subject = `${DiscussionSource.HUB}://${DiscussionType.CONTENT}/${item.id}` ``

Which sets the stage for Post.subject being an object

```ts
let subject = {
  source: DiscussionSource.HUB,
  type: DiscussionType.CONTENT,
  object: item.id
  ...
}
```

Reply:
I updated the A.C. in [2472](https://devtopia.esri.com/dc/hub/issues/2472) to remove the URI validation and flagged the `DiscussionType.ITEM` and `DiscussionType.DATASET` enum pairs for removal when v10.0.0 publishes.